res/controllers/DJ TechTools-MIDI Fighter Spectra-scripts.js

```diff
@@ -224,45 +231,36 @@ var MidiFighterSpectra;
                 });
             }
 
+            this.selectDeck = [];
             this.activeDeck = new Deck();
 
-            // Since this is a radio button set, just have one button that
-            // handles setting and unsetting all the LEDs.
-            this.selectDeck = new components.Button({
-                group: "[Channel1]",
-                key: "end_of_track",
-                midi: [0x92, [0x34, 0x54]],
-                input: function(_channel, control, value, _status, group) {
-                    MidiFighterSpectra.controller.activeDeck.setCurrentDeck(group);
-                    for (let i = 0; i < 4; i++) {
-                        for (const midino of this.midi[1]) {
-                            midi.sendShortMsg(this.midi[0], midino + i, (control === midino + i) ? this.on : this.off);
-                        }
-                    }
-                },
-                output: function(value, group, _control) {
-                    // Pulse the deck select button if the track is ending.
-                    const deckNum = parseInt(script.channelRegEx.exec(group)[1]);
-                    for (const midino of this.midi[1]) {
-                        if (value) {
-                            midi.sendShortMsg(this.midi[0] + 1, midino + deckNum - 1, 47);
-                        } else {
-                            midi.sendShortMsg(this.midi[0] - 0xF, midino + deckNum - 1, 33);
-                        }
-                    }
-                },
-                on: engine.getSetting("deckSelectedColor"),
-                off: engine.getSetting("deckUnselectedColor"),
-                connect: function() {
-                    if (engine.getSetting("pulseDeckSelect")) {
-                        for (let i = 0; i < 4; i++) {
-                            this.connections[i] = engine.makeConnection(`[Channel${i + 1}]`, "end_of_track", this.output.bind(this));
+            for (let i = 0; i < 8; i++) {
+                this.selectDeck[i] = new components.Button({
+                    group: `[Channel${(i % 4) + 1}]`,
+                    key: "end_of_track",
+                    midi: [0x92, (i < 4 ? 0x34 : 0x54) + (i % 4)],
+                    on: engine.getSetting("deckSelectedColor"),
+                    off: engine.getSetting("deckUnselectedColor"),
+                    input: function(_channel, control, value, _status, group) {
+                        MidiFighterSpectra.controller.activeDeck.setCurrentDeck(group);
+                    },
+                    setActive(group) {
```

Comment:
Is this valid syntax in the ancient ES7 we're bound to by QJSEngine? It works in nodejs, but idk about here.

Reply:
[Apparently it is](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#method_definitions). Lets stick to `property: function(...) {...}` for now, but feel free to use the shorthand notation in future mappings.

Reply:
Weird; I had no idea. Fixed for consistency either way!